apps/desktop-wallet/src/components/Ellipsed.tsx

```diff
@@ -1,49 +1,26 @@
-import { HTMLAttributes, useCallback, useEffect, useRef, useState } from 'react'
+import { HTMLAttributes } from 'react'
 import styled from 'styled-components'
 
-import { useWindowResize } from '@/utils/hooks'
-
 interface EllipsedProps extends HTMLAttributes<HTMLDivElement> {
   text: string
   className?: string
 }
 
 const Ellipsed = ({ text, className }: EllipsedProps) => {
-  const el = useRef<HTMLDivElement | null>(null)
-  const [_text, setText] = useState(text)
-
-  const handleResize = useCallback(() => {
-    if (el?.current === null) {
-      setText(text.substring(0, 5) + '...')
-    } else {
-      const visibleChars = Math.floor(el.current.clientWidth / (el.current.scrollWidth / text.length))
-      const half = visibleChars / 2
-
-      setText(
-        visibleChars >= text.length
-          ? text
-          : text.slice(0, Math.floor(half) - 2) +
-              (visibleChars === text.length ? '' : '...') +
-              text.slice(-Math.ceil(half) + 2)
-      )
-    }
-  }, [text])
-
-  useWindowResize(handleResize)
-
-  useEffect(() => {
-    handleResize()
-  }, [handleResize, text])
+  const needsEllipsis = text.length > 11
+  const truncatedText = needsEllipsis ? text.substring(0, 4) + '...' + text.slice(-4) : text
```

Comment:
So this isn't dynamic anymore. In case we display addresses where we have space (check TX modal for instance), the user will not be able to check the full hash?

Reply:
Correct, to avoid jumping around, and since we use monospace font for hashes, this PR proposes to unify hashes and display only the first and last 4 characters.